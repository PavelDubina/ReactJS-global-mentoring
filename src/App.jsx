import React from 'react'
import { hot } from 'react-hot-loader'
import { MovieList } from './components/MoviesList/MovieList'
import { Footer } from './components/Footer/Footer'
import { Header } from './components/Header/Header'
import { ErrorBoundary } from './components/ErrorBoundary/ErrorBoundary'
import { MovieDetails } from './components/MovieDetails/MovieDetails'
// eslint-disable-next-line no-unused-vars
import { Search } from './components/Search/Search'
import styles from './App.scss'

const App = () => (
  <div className={styles.main_container}>
    <Header>
      <MovieDetails />
      {/* <Search /> */}
    </Header>
    <ErrorBoundary>
      <MovieList />
    </ErrorBoundary>
    <Footer />
  </div>
)
export default hot(module)(App)
