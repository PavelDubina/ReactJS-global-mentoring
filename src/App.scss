* {
  margin: 0;
  padding: 0;
  box-sizing: border-box;
  font-family: 'Gill Sans', 'Gill Sans MT', Calibri, 'Trebuchet MS', sans-serif;
}

html,
body {
  height: 100%;
}

body > div:nth-of-type(1) {
  height: 100%;
}

::-webkit-scrollbar {
  width: 10px;
}

::-webkit-scrollbar-track {
  background: #f1f1f1;
}

::-webkit-scrollbar-thumb {
  background: #888;
}

::-webkit-scrollbar-thumb:hover {
  background: #555;
}

.main_container {
  position: relative;
  min-height: 100%;
  margin: 0 auto;
  display: flex;
  flex-direction: column;
}
