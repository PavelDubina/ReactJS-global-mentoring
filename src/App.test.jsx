import React from 'react'
import { createMemoryHistory } from 'history'
import { Provider } from 'react-redux'
import configureStore from 'redux-mock-store'
import { Router } from 'react-router-dom'
import { render } from '@testing-library/react'
import movies from './MockedData/mockMovies.json'
import movie from './MockedData/mockMovie.json'
import App from './App'

const mockStore = configureStore([])

describe('App', () => {
  const store = mockStore({ moviesData: { movies, sortBy: 'RELEASE DATE', filter: 'all' }, movieData: { movie } })
  store.dispatch = jest.fn()
  const history = createMemoryHistory()

  it('Should renders correctly with main page', () => {
    history.push('/')
    const getComponent = () =>
      render(
        <Router history={history}>
          <Provider store={store}>
            <App />
          </Provider>
        </Router>,
      )
    expect(getComponent({}).asFragment()).toMatchSnapshot()
  })
  it('Should renders correctly with search page', () => {
    history.push('/search/avengers')
    const getComponent = () =>
      render(
        <Router history={history}>
          <Provider store={store}>
            <App />
          </Provider>
        </Router>,
      )
    expect(getComponent({}).asFragment()).toMatchSnapshot()
  })
  it('Should renders correctly with movieDetails page', () => {
    history.push('/film/313369')
    const getComponent = () =>
      render(
        <Router history={history}>
          <Provider store={store}>
            <App />
          </Provider>
        </Router>,
      )
    expect(getComponent({}).asFragment()).toMatchSnapshot()
  })
})
