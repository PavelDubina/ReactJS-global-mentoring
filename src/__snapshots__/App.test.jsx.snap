// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Should renders correctly with main page 1`] = `
<DocumentFragment>
  <div
    class="main_container"
  >
    <header
      class="container"
    >
      <div
        class="container_shadow"
      />
      <section
        class="section"
      >
        <a
          href="/"
        >
          <span
            class="logo"
          >
            <b>
              netflix
            </b>
            roulette
          </span>
        </a>
        <div
          class="container"
        >
          <button
            class="adding"
            type="button"
          >
            + ADD MOVIE
          </button>
          <h1
            class="title"
          >
            FIND YOUR MOVIE
          </h1>
          <form
            class="form"
          >
            <input
              class="input"
              placeholder="What do you want to watch?"
              type="text"
              value=""
            />
            <button
              class="search"
              type="submit"
            >
              SEARCH
            </button>
          </form>
        </div>
      </section>
    </header>
    <div
      class="container"
    >
      <div
        class="section"
      >
        <nav
          class="container"
        >
          <div
            class="genres"
          >
            <div
              class="genre active"
            >
              all
            </div>
            <div
              class="genre"
            >
              documentary
            </div>
            <div
              class="genre"
            >
              comedy
            </div>
            <div
              class="genre"
            >
              horror
            </div>
            <div
              class="genre"
            >
              crime
            </div>
          </div>
          <div
            class="sort"
          >
            <span>
              SORT BY
            </span>
            <div
              class="select"
              data-testid="select"
            >
              RELEASE DATE
              <div
                class="dropdown"
                data-testid="dropdown"
              >
                <p
                  class="active"
                  data-testid="RELEASE DATE"
                >
                  RELEASE DATE
                </p>
                <p
                  class=""
                  data-testid="RATING"
                >
                  RATING
                </p>
              </div>
            </div>
          </div>
        </nav>
        <div
          class="result__count"
        >
          <span
            class="result__count_number"
          >
            1
          </span>
          <span>
            movies found
          </span>
        </div>
        <div
          class="list"
        >
          <div
            class="container"
          >
            <div
              class="poster"
            >
              <img
                alt="poster"
                height="16"
                src="https://image.tmdb.org/t/p/w500/5wioUjzwr91fvq7ydunxK2CPSxi.jpg"
                width="9"
              />
              <div
                class="menu"
                data-testid="menu"
              >
                <div
                  class="menu"
                  data-testid="card-menu"
                >
                  <div />
                  <div />
                  <div />
                </div>
              </div>
            </div>
            <div
              class="movie__data"
            >
              <div
                class="movie__data_info"
              >
                <h2
                  class="movie__data_info_title"
                >
                  Noah
                </h2>
                <p
                  class="movie__data_info_genres"
                >
                  Drama & Adventure
                </p>
              </div>
              <div
                class="movie__data_date"
              >
                <span>
                  2014
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="scroll_to_top"
      />
    </div>
    <footer
      class="container"
    >
      <span
        class="title"
      >
        <b>
          netflix
        </b>
        roulette
      </span>
    </footer>
  </div>
</DocumentFragment>
`;

exports[`App Should renders correctly with movieDetails page 1`] = `
<DocumentFragment>
  <div
    class="main_container"
  >
    <header
      class="container"
    >
      <div
        class="container_shadow"
      />
      <section
        class="section"
      >
        <a
          href="/"
        >
          <span
            class="logo"
          >
            <b>
              netflix
            </b>
            roulette
          </span>
        </a>
        <div
          class="container"
        >
          <img
            alt="poster"
            height="16"
            src="https://image.tmdb.org/t/p/w500/ylXCdC106IKiarftHkcacasaAcb.jpg"
            width="9"
          />
          <div
            class="info"
          >
            <div
              class="name_and_rating"
            >
              <h1
                class="name"
              >
                La La Land
              </h1>
              <div
                class="rating"
              >
                7.9
              </div>
            </div>
            <div
              class="duration_and_release"
            >
              <p>
                2016
              </p>
              <p
                class="runtime"
              >
                128 min
              </p>
            </div>
            <p
              class="overview"
            >
              Mia, an aspiring actress, serves lattes to movie stars in between auditions and Sebastian, a jazz musician, scrapes by playing cocktail party gigs in dingy bars, but as success mounts they are faced with decisions that begin to fray the fragile fabric of their love affair, and the dreams they worked so hard to maintain in each other threaten to rip them apart.
            </p>
          </div>
          <a
            href="/"
          >
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-search fa-w-16 search_icon"
              data-icon="search"
              data-prefix="fas"
              focusable="false"
              role="img"
              viewBox="0 0 512 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z"
                fill="currentColor"
              />
            </svg>
          </a>
        </div>
      </section>
    </header>
    <div
      class="container"
    >
      <div
        class="section"
      >
        <nav
          class="container"
        >
          <div
            class="genres"
          >
            <div
              class="genre active"
            >
              all
            </div>
            <div
              class="genre"
            >
              documentary
            </div>
            <div
              class="genre"
            >
              comedy
            </div>
            <div
              class="genre"
            >
              horror
            </div>
            <div
              class="genre"
            >
              crime
            </div>
          </div>
          <div
            class="sort"
          >
            <span>
              SORT BY
            </span>
            <div
              class="select"
              data-testid="select"
            >
              RELEASE DATE
              <div
                class="dropdown"
                data-testid="dropdown"
              >
                <p
                  class="active"
                  data-testid="RELEASE DATE"
                >
                  RELEASE DATE
                </p>
                <p
                  class=""
                  data-testid="RATING"
                >
                  RATING
                </p>
              </div>
            </div>
          </div>
        </nav>
        <div
          class="result__count"
        >
          <span
            class="result__count_number"
          >
            1
          </span>
          <span>
            movies found
          </span>
        </div>
        <div
          class="list"
        >
          <div
            class="container"
          >
            <div
              class="poster"
            >
              <img
                alt="poster"
                height="16"
                src="https://image.tmdb.org/t/p/w500/5wioUjzwr91fvq7ydunxK2CPSxi.jpg"
                width="9"
              />
              <div
                class="menu"
                data-testid="menu"
              >
                <div
                  class="menu"
                  data-testid="card-menu"
                >
                  <div />
                  <div />
                  <div />
                </div>
              </div>
            </div>
            <div
              class="movie__data"
            >
              <div
                class="movie__data_info"
              >
                <h2
                  class="movie__data_info_title"
                >
                  Noah
                </h2>
                <p
                  class="movie__data_info_genres"
                >
                  Drama & Adventure
                </p>
              </div>
              <div
                class="movie__data_date"
              >
                <span>
                  2014
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="scroll_to_top"
      />
    </div>
    <footer
      class="container"
    >
      <span
        class="title"
      >
        <b>
          netflix
        </b>
        roulette
      </span>
    </footer>
  </div>
</DocumentFragment>
`;

exports[`App Should renders correctly with search page 1`] = `
<DocumentFragment>
  <div
    class="main_container"
  >
    <header
      class="container"
    >
      <div
        class="container_shadow"
      />
      <section
        class="section"
      >
        <a
          href="/"
        >
          <span
            class="logo"
          >
            <b>
              netflix
            </b>
            roulette
          </span>
        </a>
        <div
          class="container"
        >
          <button
            class="adding"
            type="button"
          >
            + ADD MOVIE
          </button>
          <h1
            class="title"
          >
            FIND YOUR MOVIE
          </h1>
          <form
            class="form"
          >
            <input
              class="input"
              placeholder="What do you want to watch?"
              type="text"
              value="avengers"
            />
            <button
              class="search"
              type="submit"
            >
              SEARCH
            </button>
          </form>
        </div>
      </section>
    </header>
    <div
      class="container"
    >
      <div
        class="section"
      >
        <nav
          class="container"
        >
          <div
            class="genres"
          >
            <div
              class="genre active"
            >
              all
            </div>
            <div
              class="genre"
            >
              documentary
            </div>
            <div
              class="genre"
            >
              comedy
            </div>
            <div
              class="genre"
            >
              horror
            </div>
            <div
              class="genre"
            >
              crime
            </div>
          </div>
          <div
            class="sort"
          >
            <span>
              SORT BY
            </span>
            <div
              class="select"
              data-testid="select"
            >
              RELEASE DATE
              <div
                class="dropdown"
                data-testid="dropdown"
              >
                <p
                  class="active"
                  data-testid="RELEASE DATE"
                >
                  RELEASE DATE
                </p>
                <p
                  class=""
                  data-testid="RATING"
                >
                  RATING
                </p>
              </div>
            </div>
          </div>
        </nav>
        <div
          class="result__count"
        >
          <span
            class="result__count_number"
          >
            1
          </span>
          <span>
            movies found
          </span>
        </div>
        <div
          class="list"
        >
          <div
            class="container"
          >
            <div
              class="poster"
            >
              <img
                alt="poster"
                height="16"
                src="https://image.tmdb.org/t/p/w500/5wioUjzwr91fvq7ydunxK2CPSxi.jpg"
                width="9"
              />
              <div
                class="menu"
                data-testid="menu"
              >
                <div
                  class="menu"
                  data-testid="card-menu"
                >
                  <div />
                  <div />
                  <div />
                </div>
              </div>
            </div>
            <div
              class="movie__data"
            >
              <div
                class="movie__data_info"
              >
                <h2
                  class="movie__data_info_title"
                >
                  Noah
                </h2>
                <p
                  class="movie__data_info_genres"
                >
                  Drama & Adventure
                </p>
              </div>
              <div
                class="movie__data_date"
              >
                <span>
                  2014
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="scroll_to_top"
      />
    </div>
    <footer
      class="container"
    >
      <span
        class="title"
      >
        <b>
          netflix
        </b>
        roulette
      </span>
    </footer>
  </div>
</DocumentFragment>
`;
